test/merge-patient-lab-iips-mocha.js

```diff
@@ -5,151 +5,260 @@ var chaiAsPromised = require('chai-as-promised');
+
+                            // initialize state
+                            component.rpf.vni('').inputState('patient',
+                                                             { id: '001', 
+                                                               name: 'Alice', 
+                                                               dob: '1979-01-23' });
+
+                            var state = component.rpf.vni('').inputState('patient');
+                            state.should.be.an('object');
+                            state.should.have.all.keys( 'data', 'lm', 'previousLms', 'port' );
+                    });
+                });
+
+                it("should have input state data after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            var patientState = { id: '001', name: 'Alice', dob: '1979-01-23' };
+                            component.rpf.vni('').inputState('patient',
+                                                             patientState );
+
+                            var state = component.rpf.vni('').inputState('patient');
+                            state.should.be.an('object');
+			    state.data.should.deep.equal( patientState );
+                    });
+                });
             });
-        });
-        describe('#getOutputState', function() {
-            it("should initially return nothing", function() {
-                return Promise.resolve(createComponent()).then(function(component){
-                    return component.rpf.createVni('').getOutputState();
-                }).should.become(undefined);
+
+            describe('#outputState', function() {
+
+                it("should initially return nothing", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+                            return component.rpf.vni('').outputState();
+                    }).should.become(undefined);
+                });
+
+                it("should have a output state after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            component.rpf.vni('').outputState( { id: '001', 
+                                                                 name: 'Alice', 
+                                                                 dob: '1979-01-23', 
+                                                                 glucose: '75',  
+                                                                 date: '2012-02-01'});
```

Comment:
Again, this should be a full State object, with data and lm properties -- not just the data part.